components/esp_modem/examples/modem_tcp_client/components/extra_tcp_transports/include/mbedtls_wrap.hpp

```diff
@@ -39,7 +44,9 @@ class Tls {
     mbedtls_ssl_config conf_{};
     mbedtls_ctr_drbg_context ctr_drbg_{};
     mbedtls_entropy_context entropy_{};
+    std::unique_ptr<mbedtls_ssl_session> session_{nullptr};
     virtual void delay() {}
+    bool session(session_op op);
```

Comment:
Have you considered adding different functions instead of the usage of the enum? 
```suggestion
    bool set_session();
    bool get_session();
    void reset_session();
    bool session_loaded();
```

Also, from the documentation I think store and load are better names than get and set, but I understand the usage of the latter from mbedtls names.

Reply:
Good point, thanks for suggesting. This looks more "programatically" correct. Just changed `session_loaded()` to `is_session_loaded()`